code/game/gamemodes/changeling/changeling.dm

```diff
@@ -277,18 +281,19 @@ var/list/possible_changeling_IDs = list("Alpha","Beta","Gamma","Delta","Epsilon"
 
 
 /datum/changeling/proc/get_dna(dna_owner)
-	for(var/datum/dna/DNA in (absorbed_dna+protected_dna))
-		if(dna_owner == DNA.real_name)
-			return DNA
+	for(var/datum/changelingprofile/prof in stored_profiles)
+		if(dna_owner == prof.name)
+			return prof
 
 /datum/changeling/proc/has_dna(datum/dna/tDNA)
-	for(var/datum/dna/D in (absorbed_dna+protected_dna))
-		if(tDNA.is_same_as(D))
+	for(var/datum/changelingprofile/prof in stored_profiles)
+		if(tDNA.is_same_as(prof.dna))
 			return 1
 	return 0
 
-/datum/changeling/proc/can_absorb_dna(mob/living/carbon/user, mob/living/carbon/target)
-	if(absorbed_dna[1] == user.dna)//If our current DNA is the stalest, we gotta ditch it.
+/datum/changeling/proc/can_absorb_dna(mob/living/carbon/user, mob/living/carbon/human/target)
+	var/datum/changelingprofile/prof = stored_profiles[1]
+	if(prof.dna == user.dna)//If our current DNA is the stalest, we gotta ditch it.
 		user << "<span class='warning'>We have reached our capacity to store genetic information! We must transform before absorbing more.</span>"
 		return

```

Comment:
This check is true for lings that haven't absorbed any dna yet (and maybe other cases). This means you can never absorb dna and always get "We have reached our capacity to store genetic information! " with every form of dna absorption.